packages/perseus-editor/src/widgets/label-image-editor.tsx

```diff
@@ -151,6 +151,14 @@ class LabelImageEditor extends React.Component<Props> {
         if (url) {
             Util.getImageSize(url, (width, height) => {
                 this.props.onChange({
+                    /**
+                     * Sending `imageUrl` up again
+                     * (even though we did so at the beginning of handleImageChange)
+                     * because we ran into a race condition (LEMS-2583) where
+                     * `imageUrl` was getting set to an empty string if measuring
+                     * happened too fast.
+                     */
+                    imageUrl: url,
```

Comment:
I wonder... is it correct that we're firing the `url` before we know the image `height` and `width` at all? If we already had a URL that we measured, we get inconsistent data by firing the URL by itself before measurement completes. 🤔 I suspect this is how it's been for a long time, but just something to consider for future.